src/Text/Pandoc/App/CommandLineOptions.hs

```diff
@@ -529,15 +553,22 @@ options =
                  "" -- "Path of custom reference doc"
 
     , Option "" ["self-contained"]
-                 (NoArg
-                  (\opt -> do
+                 (OptArg
+                  (\arg opt -> do
                     deprecatedOption "--self-contained" "use --embed-resources --standalone"
-                    return opt { optSelfContained = True }))
+                    case readBoolFromOptArg arg of 
+                           Just boolValue -> return opt { optSelfContained = boolValue }
+                           Nothing -> optError $ PandocOptionError "value must be either true or false")
+                    "[true|false]")
```

Comment:
There's a lot of code repetition here.  Why not have
```
readBoolFromOptArg :: String -> ExceptT OptInfo IO Bool
```
(so that it can raise the error directly instead of just returning a Maybe value) and then this code can be simplified to
```hs
boolValue <- readBoolFromOptArg arg
return opt { opthSelfContained = boolValue }
```